core/src/main/java/org/apache/iceberg/view/ViewMetadataParser.java

```diff
@@ -153,12 +153,12 @@ public static ViewMetadata fromJson(String metadataLocation, JsonNode json) {
         formatVersion,
         location,
         schemas,
+        properties,
+        metadataLocation,
         currentVersionId,
         versions,
         historyEntries,
-        properties,
```

Comment:
With Interface `IcebergMetadata` which has common members from TableMetadata and ViewMetadata . `ImmutableViewMetadata` has different order in constructors.

Reply:
Does the changing order break something?

Reply:
As a natural order it was breaking all the IMPL of ViewMetadata. Since it was considering members of parent class first. 
After adding order no explicitly as `@Value.Parameter(order = 1001)` . It became backward compatible. 

I think my above comment is irrelevant now.